### DISCUSSION TITLE
An agent spawn waiting queue for 2-dimensional movement agent in traffic intersection case study

### DISCUSSION THREAD

**Discussion by mrak13** [mesa/mesa#1359] [Q&A] [answered]
![Q-A mesa](https://user-images.githubusercontent.com/38686397/172763022-d26bf358-1227-4799-801d-394f6c89d939.jpg)

Greetings fellow modellers.
I have developed a traffic intersection case study using MESA where the agent learns using Q-learning to avoid collision with other agents.
When the agent collides with other agents or reaches an 'exit point' (grid with red dot), the agent will respawn to the respawn point (grid with blue dot) at random. I'm opting to respawn agents instead of removing agents from the environments to keep the individual agent's memory/experience (q-tables - where the agent keep its rewards/punishment outcomes).

Problem: I'm having some issues with the agent colliding at the 'spawn point' - hence the 'spawn  point congestion' problem. Is there a solution where i can maybe put the agent in a temporary queue and 'moveto' agent to the spawn point when the 'spawn point' grid is empty?

**Comment by rht** [comment, marked as answer]
A quick hack: you can mark the agent state to be "in queue", which you can set it so that it prevents them from being in collision with other agents. You remove the agent from the grid, but keep them in the scheduler. When it's the agent's turn in the queue, you can put them back into the grid at the respawn point.

**Reply by mrak13** [reply]
It works. I made a list of agent 'InQueue' that keeps the agent instance name to access and releases the agent on empty spawn points. However need to note that 'move_to' doesnt work got an error due to 'nonetype' error since 'move_agent' requires current position coordinates, so used 'place_agent'

Thanks
